runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineOptions.java

```diff
@@ -48,6 +48,25 @@
   Long getBatchIntervalMillis();
   void setBatchIntervalMillis(Long batchInterval);
 
+  @Description("A checkpoint directory for streaming resilience, ignored in batch. "
+      + "For durability, a reliable filesystem such as HDFS/S3/GS is necessary.")
+  @Default.InstanceFactory(CheckpointDirFactory.class)
+  String getCheckpointDir();
+  void setCheckpointDir(String checkpointDir);
+
+  /**
+   * Returns the default checkpoint directory of /tmp/${app.name}-${start_time_millis}.
+   * For testing purposes only. Production applications should use a reliable
```

Comment:
I will log in case of local filesystem, I don't see a reason to add yet another `option` for this. 
BTW is there a way to write "validator" for options (like there is default factory..) ?

This is only valid in streaming, batch will simply restart.